corehq/apps/locations/models.py

```diff
@@ -254,6 +311,12 @@ def _sync_location(self):
             }
         )
 
+        if is_new or (sql_location.location_type.name != self.location_type):
+            sql_location.location_type, _ = LocationType.objects.get_or_create(
+                domain=self.domain,
+                name=self.location_type,
+            )
```

Comment:
I had actually corrected this, but that's on a different branch (the one with the non-null constraint) was having some trouble getting it past the tests, which create locations without types left and right.